src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java

```diff
@@ -162,7 +162,19 @@ public EC2AbstractSlave(String name, String instanceId, String templateDescripti
         this.amiType = amiType;
         this.maxTotalUses = maxTotalUses;
         readResolve();
-        fetchLiveInstanceData(true);
+        try {
+            // Wait up to 1 minute for the instance to show up
+            fetchLiveInstanceData(true, 60);
```

Comment:
When does this get called? If its during a provision operation, having such a long timeout might not be ideal as during provisioning the queue lock is held which means during this 1 minute builds you start won't be queued up and builds can't be picked up by nodes.

Reply:
I'm not totally sure what Jenkins context this is called in, but I'm pretty sure it doesn't block other builds from queuing. That said, it already blocks for up to 25 seconds. This just extends it to 60 seconds.

We could try leaving out this commit and just having the commit that tries to terminate the instance if it couldn't get the information about the instance. That would hopefully keep it from getting deadlocked.

Yet another possible fix is for the EC2 plugin to keep track of which instances it launched and actively terminate any other instances it finds in EC2 with the same details. I didn't look into that, though.

Reply:
Could we keep the timeout to be the same and explore increasing it if it does not solve the issue completely